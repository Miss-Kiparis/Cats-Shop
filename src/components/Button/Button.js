import "./Button.scss";
import PropTypes from "prop-types";

function Button(props) {
  return (
    <button
      data-testid="btn"
      className={props.btnStyles}
      type="button"
      onClick={props.btnAction}
    >
      {props.text}
    </button>
  );
}

Button.defaultProps = {
  type: "button",
  btnAction: () => {},
  btnStyles: "btn__bad",
};

Button.propTypes = {
  btnStyles: PropTypes.string,
  type: PropTypes.string,
  btnAction: PropTypes.func,
  text: PropTypes.string,
};

export default Button;
