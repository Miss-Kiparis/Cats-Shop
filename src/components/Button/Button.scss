
.cart__btn {
    padding: 15px 20px;
    background: #fff;
    text-transform: capitalize;
    font-family: ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;

    border: none;
    border-radius: 50px;
    color: #000;
    outline: none;
    cursor: pointer;
    transition: all 0.4s;
   }
   
   .cart__btn:hover {
    box-shadow:  inset 0 0 0 3px #ffd166;
    background: #ffd166;
    color: #fff;
   }

// using in Modal.js
.btn__container {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 350px;
    margin: 0 auto;
}

.btn__good {
    padding: 15px 20px;
    background: #fff;
    text-transform: capitalize;
    font-family: ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;

    border: none;
    border-radius: 50px;
    color: #000;
    outline: none;
    cursor: pointer;
    transition: all 0.4s;

        &:hover {
            box-shadow: 
            
            inset 0 0 0 10px #3fd476;
            color: #3fd476;
    }
}

.btn__bad {
    padding: 15px 20px;
    background: #fff;
    text-transform: capitalize;
    font-family: ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;

    border: none;
    border-radius: 50px;
    color: #000;
    outline: none;
    cursor: pointer;
    transition: all 0.4s;

        &:hover {
            box-shadow: 
            
            inset 0 0 0 10px #e4341a;
            color: #e4341a;
    }
}