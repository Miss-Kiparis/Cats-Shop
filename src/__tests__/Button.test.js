import "@testing-library/jest-dom";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import Button from "../components/Button/Button";
import renderer from "react-test-renderer";

describe("Button", () => {
  afterEach(() => {
    cleanup();
  });

  const mockData = {
    text: "mock text",
    btnStyles: "mock_style",
    btnAction: jest.fn(),
  };

  test("should exist", () => {
    render(
      <Button
        btnStyles={mockData.btnStyles}
        text={mockData.text}
        btnAction={mockData.btnAction}
      />
    );
    const btn = screen.getByTestId("btn");
    expect(btn).toBeInTheDocument();
  });

  test("should have some text", () => {
    render(
      <Button
        btnStyles={mockData.btnStyles}
        text={mockData.text}
        btnAction={mockData.btnAction}
      />
    );
    const btn = screen.getByTestId("btn");
    expect(btn).toHaveTextContent("mock text");
  });

  test("should have btnAction", () => {
    render(
      <Button
        btnStyles={mockData.btnStyles}
        text={mockData.text}
        btnAction={mockData.btnAction}
      />
    );
    const btn = screen.getByTestId("btn");
    fireEvent.click(btn);
    expect(mockData.btnAction).toHaveBeenCalled();
  });
});

test("matches snapshot", () => {
  const mockData = {
    text: "mock text",
    btnStyles: "mock_style",
    btnAction: jest.fn(),
  };
  const tree = renderer
    .create(
      <Button
        btnStyles={mockData.btnStyles}
        text={mockData.text}
        btnAction={mockData.btnAction}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
