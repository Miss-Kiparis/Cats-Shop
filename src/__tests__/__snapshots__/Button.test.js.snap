// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="mock_style"
  data-testid="btn"
  onClick={[MockFunction]}
  type="button"
>
  mock text
</button>
`;
